src/rust/runtime/mod.rs

```diff
@@ -107,37 +109,35 @@ impl SharedDemiRuntime {
     #[cfg(test)]
     pub fn new(now: Instant) -> Self {
         timer::global_set_time(now);
+        let mut scheduler: SharedScheduler = SharedScheduler::default();
+        let foreground_group_id: TaskId = scheduler.create_group();
+        let background_group_id: TaskId = scheduler.create_group();
         Self(SharedObject::<DemiRuntime>::new(DemiRuntime {
             qtable: IoQueueTable::default(),
-            scheduler: SharedScheduler::default(),
+            scheduler,
+            foreground_group_id,
+            background_group_id,
             socket_id_to_qdesc_map: SocketIdToQDescMap::default(),
             ts_iters: 0,
             completed_tasks: HashMap::<QToken, (QDesc, OperationResult)>::new(),
         }))
     }
 
-    /// Inserts the `coroutine` named `task_name` into the scheduler.
-    pub fn insert_io_coroutine<F: FusedFuture<Output = (QDesc, OperationResult)> + 'static>(
+    /// Inserts the background `coroutine` named `task_name` into the scheduler. There should only be one of these
+    /// because we should never be polling with more than one coroutine.
+    pub fn insert_io_polling_coroutine<F: FusedFuture<Output = ()> + 'static>(
         &mut self,
         task_name: &'static str,
         coroutine: Pin<Box<F>>,
     ) -> Result<QToken, Fail> {
-        self.insert_coroutine(task_name, coroutine)
-    }
-
-    /// Inserts the background `coroutine` named `task_name` into the scheduler
-    pub fn insert_background_coroutine<F: FusedFuture<Output = ()> + 'static>(
-        &mut self,
-        task_name: &'static str,
-        coroutine: Pin<Box<F>>,
-    ) -> Result<QToken, Fail> {
-        self.insert_coroutine(task_name, coroutine)
+        self.insert_coroutine(task_name, Some(self.background_group_id), coroutine)
     }
 
     /// Inserts a coroutine of type T and task
     pub fn insert_coroutine<F: FusedFuture + 'static>(
         &mut self,
         task_name: &'static str,
+        group_id: Option<TaskId>,
```

Comment:
The value passed into this parameter is `None` almost every time times (except for io_polling coroutines). We should create a separate fn that does not accept this parameter at all to avoid ambigiuty. e.g., `insert_io_coroutine(self, task_name, coroutine)`.

Reply:
done